### PR TITLE
Tidy away `BorrowedPlainMessage`

### PR DESCRIPTION
**Pull request by ctz**

### CODE REVIEW

File: rustls/tests/api.rs [hunk 1]
```diff
@@ -798,21 +787,23 @@ fn test_tls13_late_plaintext_alert() {
...
-        ))
+        .read_tls(&mut io::Cursor::new(&build_alert(AlertLevel::Fatal, AlertDescription::UnknownCA, &[])))
         .unwrap();
 
     // The server should produce a decrypt error, trying to decrypt a plaintext alert.
     assert_eq!(server.process_new_packets(), Err(Error::DecryptError));
 }
 
+fn build_alert(level: AlertLevel, desc: AlertDescription, suffix: &[u8]) -> Vec<u8> {
+    let mut v = vec![ ContentType::Alert.get_u8() ];
```
Comment (djc) [inline rustls/tests/api.rs:798]: Not sure why were not using `Message::build_alert()` here anymore?

**Review by djc — APPROVED** [review]
Nice!

File: rustls/tests/api.rs (hunk 1 above)
Comment (ctz) [reply rustls/tests/api.rs:798]: `Message::build_alert` can only make syntactically valid messages, but this function is less restrictive. An additional side-effect of this change is to remove the only use of `to_unencrypted_opaque()` on `InboundMessage`.

Comment (djc) [reply rustls/tests/api.rs:798]: Fair!

**Review by cpu — APPROVED** [review]
:broom: :+1:

**Pull request merged**